module/VuFind/src/VuFind/Recommend/PluginManager.php

```diff
@@ -124,7 +124,7 @@ class PluginManager extends \VuFind\ServiceManager\AbstractPluginManager
         RecommendLinks::class => InjectConfigManagerFactory::class,
         RemoveFilters::class => InvokableFactory::class,
         SideFacets::class => SideFacetsFactory::class,
-        SideFacetsDeferred::class => InjectConfigManagerFactory::class,
+        SideFacetsDeferred::class => SideFacetsFactory::class,
```

Comment:
It was actually using wrong factory. Since SideFacetsDeferred descends from SideFacets, it should use the correct factory as well. Wrong factory meant that the class didn't get HierarchicalFacetHelper passed to its constructor, and that caused getFacetSet to balk with hierarchical facets. I'm not sure why this hasn't come up earlier.

Reply:
That is strange! I thought I had tested all of that functionality at the time we added this and it was working. My understanding was that most of the logic took place in the AJAX helper, which is why the SideFacetsDeferred module itself didn't need all of its dependencies... but that may well be incorrect.

In any case, if this is a bug fix, maybe we should commit it back to the dev branch so it can be corrected before we get through this process.

Reply:
I'll dig a bit more to find out when this is called. May be something that only affects Blender for some reason.

Reply:
It's also entirely possible that my testing was not sufficiently thorough -- since I don't use hierarchical facets locally, I may have overlooked something.

Reply:
Aha! it was the newly added code that disabled a checkbox facet and needed to call getFacetSet() to accomplish that. Latest version doesn't use it anymore, but this should probably be picked separately as the fix is correct anyway.

Reply:
I've extracted this to #2347, if you have a moment to double-check and approve.